src/utils/chartOptionsHelpers.js

```diff
@@ -81,7 +80,7 @@ export const plotNullDataLayerToAxis = (xAxis, series, broadcastSetState) => {
         events: {
           mouseover: function(e) {
             broadcastSetState &&
-              broadcastSetState({'customLegend': createCartesianCustomLegendData(this.axis.series, null)});
+              broadcastSetState({'customLegend': createCartesianCustomLegendData(this.axis.series, idx)});
```

Comment:
also, not sure this will work if the null data layer is not in first position, as in series Jan - Sept, null at August